### PR TITLE
prevent dia/bio dots from stacking

### PR DESCRIPTION
**Pull request by Brierre**
**_I affirm:_**
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] that I agree to LandSandBoat's [Limited Contributor License Agreement](https://github.com/LandSandBoat/server/blob/base/.github/CONTRIBUTOR_AGREEMENT.md), as written on this date
- [x] that I have **read the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)**
- [x] that I've _**tested my code and things my code changed**_ since the last commit in the PR, and will test after any later commits

### CODE REVIEW

File: scripts/globals/spells/dia.lua [hunk 1]
```diff
@@ -46,15 +46,19 @@ spell_object.onSpellCast = function(caster, target, spell)
...
             target:delStatusEffect(xi.effect.BIO)
         end
+    else
+        -- Do it!
+        if bio ~= nil then -- if mob has bio effect
+            spell:setMsg(xi.msg.basic.MAGIC_DMG) -- hit for initial damage, but no dot effects
+        else -- otherwise hit for initial damage and add dot
+            target:addStatusEffect(xi.effect.DIA, 1 + dotBonus, 3, duration, 0, 10, 1)
+            spell:setMsg(xi.msg.basic.MAGIC_DMG)
+        end
```
Comment (Xaver-DaRed) [inline scripts/globals/spells/dia.lua]: This probably applies to all, changed files, but for this particular one...
I would move the setmsg line ouside of the if/else check.
that would make the bio check empty, so, that would have to be edited aswell.

So maybe something like...

if bio == nil then
    target:addStatusEffect(xi.effect.DIA, 1 + dotBonus, 3, duration, 0, 10, 1)
end
spell:setMsg(xi.msg.basic.MAGIC_DMG)

File: scripts/globals/spells/dia.lua [hunk 2]
```diff
@@ -46,15 +46,19 @@ spell_object.onSpellCast = function(caster, target, spell)
...
-    -- Do it!
-    target:addStatusEffect(xi.effect.DIA, 1 + dotBonus, 3, duration, 0, 10, 1)
-    spell:setMsg(xi.msg.basic.MAGIC_DMG)
-
     -- Try to kill same tier Bio (non-default behavior)
     if BIO_OVERWRITE == 1 and bio ~= nil then
         if bio:getPower() == 1 then
             target:delStatusEffect(xi.effect.BIO)
         end
+    else
```
Comment (Xaver-DaRed) [inline scripts/globals/spells/dia.lua]: If this allows overwrite, then it deletes proper bio tier... but then it does nothing else, not even damage.

File: scripts/globals/spells/dia.lua (hunk 1 above)
Comment (Brierre) [reply scripts/globals/spells/dia.lua]: changed this to clean things up

File: scripts/globals/spells/dia.lua (hunk 2 above)
Comment (Brierre) [reply scripts/globals/spells/dia.lua]: And moved this around to make it do what it was supposed to do.

**Review by Xaver-DaRed — APPROVED** [review]
LGTM

**Comment by Xaver-DaRed** [comment]
EDITED: TYVM

**Pull request merged**